### ISSUE TITLE
Add support for bypassing branch protection to sync-repo-settings bot

### ISSUE THREAD

**Issue by sararob** [googleapis/repo-automation-bots#4617]
The googlapis/python-aiplatform repo is using [Copybara](https://github.com/google/copybara) as part of our contribution workflow. In order to merge pull requests, the Copybara app needs to be able to bypass branch protections. I think this is a new setting on GitHub.

Could the `sync-repo-settings` bot add a [config option here](https://github.com/googleapis/repo-automation-bots/blob/main/packages/sync-repo-settings/src/schema.json) to enable unchecking the box for "Do not allow bypassing the above settings" in GitHub's branch protection rules? This will allow only the Copybara app and repo admins to bypass these settings.

**Comment by chingor13** [comment]
I see an API call for this: https://docs.github.com/en/rest/branches/branch-protection#set-app-access-restrictions, but when I query the endpoint, I don't see existing entries (it's possibly hidden by a feature flag header for beta features that I don't see documented anywhere).

**Comment by sararob** [comment]
Thanks for looking into this! Can you clarify what you mean by existing entries? For our repo, we'd want to set this as:

```
"bypass_pull_request_allowances":{"apps":["copybara-service"]}
```

**Comment by chingor13** [comment]
Yes, we can possibly set it, but the getter API does not seem to return what the documentation states. For example:

```
curl -H "Accept: application/vnd.github+json" -H "Authorization: Bearer ${GITHUB_TOKEN}" https://api.github.com/repos/googleapis/python-aiplatform/branches/main/protection/restrictions/apps
{
  "message": "Push restrictions not enabled",
  "documentation_url": "https://docs.github.com/rest/reference/repos#list-apps-with-access-to-the-protected-branch"
}
```

I would expect the API to return the current allowlisted app in that response. I have little confidence that this API is actually working.

**Comment by sararob** [comment]
Hmm, what does a GET request to `https://api.github.com/repos/googleapis/python-aiplatform/branches/main/protection` return for our repo?

If setting `bypass_pull_request_allowances` isn't possible right now, is there a way to disable the `sync-repo-settings` bot (and not use the default settings) until we can set this?

**Comment by chingor13** [comment]
> what does a GET request to `https://api.github.com/repos/googleapis/python-aiplatform/branches/main/protection` return for our repo?

When I ran the GET request it returned that `Push restrictions not enabled` response which was unexpected because I see it in the UI.

Changing this to a feature request as #4626 implemented a workaround to skip stomping branch protection

**Comment by ZhenyiQ** [comment]
@chingor13 We also have a use case for this feature. Can this have another look from someone on the team?

**Comment by ZhenyiQ** [comment]
It seems that it's this setting ("Do not allow bypassing the above checkings") that gets checked each time the sync-repo-setting bot syncs the settings: https://screenshot.googleplex.com/84jjXPSxZxNSULo

The "Allow specified actors to bypass required pull requests" setting is unaffected: https://screenshot.googleplex.com/3EfFu5eP5868znw (Copybara is always allowed to do this but it can't bypass the checkings).

Here is the settings: https://github.com/googleapis/google-cloud-java/blob/main/.github/sync-repo-settings.yaml

**Comment by ZhenyiQ** [comment]
Actually I'm wondering if this (isAdminEnforced) is the same as the "Do not allow bypassing the above settings": https://screenshot.googleplex.com/yTUCBsYP5k2uaaS
